physics/forkable_body.hpp

```diff
@@ -153,6 +153,16 @@ not_null<Tr4jectory*> Forkable<Tr4jectory, It3rator>::root() {
 }
 
 template<typename Tr4jectory, typename It3rator>
+Tr4jectory const* Forkable<Tr4jectory, It3rator>::parent() const {
+  return parent_;
```

Comment:
In Trajectory we generally check invariants and not return some vaguely defined default.  Given that the sole client doesn't check for nullness, then yes, this should return not_null<>.  Calling this function on a nonroot is a serious bug as things are written.